lib/ansible/modules/source_control/gitlab/gitlab_project_variable.py

```diff
@@ -138,18 +154,19 @@ def get_project(self, project_name):
     def list_all_project_variables(self):
         return self.project.variables.list()
 
-    def create_variable(self, key, value):
+    def create_variable(self, key, value, masked, protected):
         if self._module.check_mode:
             return
-        return self.project.variables.create({"key": key, "value": value})
+        return self.project.variables.create({"key": key, "value": value,
+                                              "masked": masked, "protected": protected})
```

Comment:
Does this work with every version of `python-gitlab` that supports variables?

Reply:
hm I don't know. I have to research.

Reply:
python-gitlab version doesn' matter.  
`project.variables.create()` is just a wrapper of a path post command. 

https://github.com/python-gitlab/python-gitlab/blob/master/gitlab/__init__.py#L676

so even when they introduce new attributes, they must not be implemented in the python-gitlab sdk.  
the used gitlab host (gitlab version) must support `masked` and `protected` those attributes.

Reply:
```
Masked variables

Introduced in GitLab 11.10
```

```
Protected environment variables

Introduced in GitLab 9.3.
```

@felixfontein do you think we need to cover this? But I don't know if old gitlab throws an error when they see attributes they don't know or if they just ignore them.

Reply:
```
>>> project.variables.create({"key": 'testc', "value": "value", "somethingelse": True})
<ProjectVariable key:testc>
>>> 
```

I guess GitLab simply ignores attribute which it doesn't know. But I add some version hints to the documentation.

Reply:
You could simply only pass variables that are `True`, then you're on the safe side :) I.e. something like
```.py
        params = {"key": key, "value": value}
        if masked:
            params["masked"] = True
        if protected:
            params["protected"] = True
        return self.project.variables.create(params)
```